packages/main/src/plugin/index.ts

```diff
@@ -1097,9 +1097,15 @@ export class PluginSystem {
     this.ipcHandle(
       'container-provider-registry:pushImage',
       async (_listener, engine: string, imageId: string, callbackId: number): Promise<void> => {
-        return containerProviderRegistry.pushImage(engine, imageId, (name: string, data: string) => {
-          this.getWebContentsSender().send('container-provider-registry:pushImage-onData', callbackId, name, data);
-        });
+        const msgName = 'container-provider-registry:pushImage-onData';
+        return containerProviderRegistry
+          .pushImage(engine, imageId, (name: string, data: string) => {
+            this.getWebContentsSender().send(msgName, callbackId, name, data);
+          })
+          .catch((error: unknown) => {
+            this.getWebContentsSender().send(msgName, callbackId, 'data', JSON.stringify({ error: String(error) }));
```

Comment:
question: why the error would be on the data channel and not on the error channel ?
usually we're sending the data on a specific onError handler

`JSON.stringify({ error: String(error) })` looks like a little bit overkill to me vs providing the `String(error)` into the error callback

Reply:
fixed renderer and main side